src/NuGet.Clients/VsExtension/VisualStudioCredentialProvider.cs

```diff
@@ -89,7 +89,7 @@ public VisualStudioCredentialProvider(IVsWebProxy webProxyService)
                 // or request credentials. 
                 WebRequest.DefaultWebProxy = new WebProxy(uriToDisplay);
 
-                return PromptForCredentials(uri, cancellationToken);
+                return Task.FromResult(PromptForCredentials(uri, cancellationToken));
```

Comment:
That just moves the problem upstream, right? PromptForCredentials would then need Task.FromResult calls (and there would be two of them instead of one) for the CredentialResponse constructors. I don't understand how this is an improvement.